### PR TITLE
Feature/backend/user setting again

### PR DESCRIPTION
**Pull request by gdchamal**
API to manage user settings
- GET /settings
- PATCH /settings

Settings are initialized when an user is created

### CODE REVIEW

File: src/backend/main/py.main/caliopen_main/user/core/user.py [hunk 1]
```diff
@@ -334,6 +348,29 @@ def setup_system_tags(self):
             tag['date_insert'] = datetime.datetime.now(tz=pytz.utc)
             Tag.create(self, **tag)
 
+    def setup_settings(self):
+        """Create settings related to user."""
+        # XXX set correct values
```
Comment (sapiens-sapide) [inline src/backend/main/py.main/caliopen_main/user/core/user.py]: We will need to fill these defaults settings from config file.

File: src/backend/main/py.main/caliopen_main/user/objects/settings.py [hunk 2]
```diff
@@ -32,5 +32,5 @@ class Settings(ObjectUser):
     }
 
     _model_class = ModelSettings
-    _pkey_name = 'device_id'
+    _pkey_name = None
```
Comment (sapiens-sapide) [inline src/backend/main/py.main/caliopen_main/user/objects/settings.py]: I don't understand why you are removing the _pkey_name attribute. «Settings» is an `ObjectUser` ; it belongs to an user and has `user_id` as its primary key.

File: src/backend/interfaces/REST/py.server/caliopen_api/user/settings.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    @view(renderer='json', permission='authenticated')
+    def get(self):
+        """Return user settings."""
+        objects = ObjectSettings.list_db(self.user.user_id)
+        settings = [x.marshall_dict() for x in objects]
+        return settings[0]
+
+    @view(renderer='json', permission='authenticated')
+    def patch(self):
+        """Update settings with payload.
```
Comment (sapiens-sapide) [inline src/backend/interfaces/REST/py.server/caliopen_api/user/settings.py]: comments are copied from 'message'… ;-)

File: src/backend/main/py.main/caliopen_main/user/core/user.py (hunk 1 above)
Comment (gdchamal) [reply src/backend/main/py.main/caliopen_main/user/core/user.py]: yes

**Pull request merged**